upgrade/upgrades: TestSystemPrivilegesUserIDMigration1500Users failed

upgrade/upgrades.TestSystemPrivilegesUserIDMigration1500Users [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11390472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11390472?buildTab=artifacts#/) on master @ [9fcc5c8ce60d2f25e06e41bd97e20445233f9fa5](https://github.com/cockroachdb/cockroach/commits/9fcc5c8ce60d2f25e06e41bd97e20445233f9fa5):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 132742 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
I230819 02:14:50.897917 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9780  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:50.901857 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9781  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:50.902924 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9782  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:51.720295 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9783  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:51.724181 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9784  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:51.725114 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9785  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:52.530040 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9786  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:52.534814 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9787  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:52.535857 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9788  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:53.123481 36360 2@util/log/event_log.go:32  [T1,Vsystem,n1] 9789 ={"Timestamp":1692411293123477399,"EventType":"runtime_stats","MemRSSBytes":1643393024,"GoroutineCount":594,"MemStackSysBytes":9043968,"GoAllocBytes":1002197624,"GoTotalBytes":1351004712,"GoStatsStaleness":0.001388283,"HeapFragmentBytes":42151304,"HeapReservedBytes":216432640,"HeapReleasedBytes":1048576,"CGoAllocBytes":142915872,"CGoTotalBytes":261029888,"CGoCallRate":67900.195,"CPUUserPercent":102.899536,"CPUSysPercent":5.799974,"GCPausePercent":0.0011787646,"GCRunCount":284,"NetHostRecvBytes":497769424,"NetHostSendBytes":497769424}
I230819 02:14:53.123611 36360 2@server/status/runtime_log.go:47  [T1,Vsystem,n1] 9790  runtime stats: 1.5 GiB RSS, 594 goroutines (stacks: 8.6 MiB), 956 MiB/1.3 GiB Go alloc/total(stale) (heap fragmentation: 40 MiB, heap reserved: 206 MiB, heap released: 1.0 MiB), 136 MiB/249 MiB CGO alloc/total (67900.2 CGO/sec), 102.9/5.8 %(u/s)time, 0.0 %gc (284x), 475 MiB/475 MiB (r/w)net
I230819 02:14:53.365506 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9791  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:53.369832 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9792  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:53.370751 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9793  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:54.183679 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9794  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:54.187515 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9795  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:54.188422 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9796  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:55.011677 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9797  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:55.015751 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9798  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:55.016708 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9799  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:55.847878 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9800  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:55.851824 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9801  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:55.852758 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9802  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:56.679275 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9803  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:56.683321 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9804  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:56.684346 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9805  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:56.684463 36794 3@pebble/event.go:721  [n1,s1,pebble] 9806  [JOB 46] WAL created 000032 (recycled 000014)
I230819 02:14:56.684600 133248 3@pebble/event.go:681  [n1,s1,pebble] 9807  [JOB 47] flushing 1 memtable (64MB) to L0
I230819 02:14:56.684665 133248 3@pebble/event.go:701  [n1,s1,pebble] 9808  [JOB 47] flushing: sstable created 000033
I230819 02:14:56.739000 133248 3@pebble/event.go:685  [n1,s1,pebble] 9809  [JOB 47] flushed 1 memtable (64MB) to L0 [000033] (676KB), in 0.1s (0.1s total), output rate 12MB/s
I230819 02:14:56.739146 133508 3@pebble/event.go:669  [n1,s1,pebble] 9810  [JOB 49] compacting(default) L0 [000033] (676KB) + L6 [000031] (2.6MB)
I230819 02:14:56.739561 133508 3@pebble/event.go:701  [n1,s1,pebble] 9811  [JOB 49] compacting: sstable created 000034
I230819 02:14:56.781935 133508 3@pebble/event.go:673  [n1,s1,pebble] 9812  [JOB 49] compacted(default) L0 [000033] (676KB) + L6 [000031] (2.6MB) -> L6 [000034] (2.6MB), in 0.0s (0.0s total), output rate 61MB/s
I230819 02:14:56.782083 36409 3@pebble/event.go:705  [n1,s1,pebble] 9813  [JOB 49] sstable deleted 000031
I230819 02:14:56.782118 36409 3@pebble/event.go:705  [n1,s1,pebble] 9814  [JOB 49] sstable deleted 000033
I230819 02:14:57.512915 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9815  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:57.516658 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9816  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:57.517403 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9817  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:58.330980 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9818  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:58.334514 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9819  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:58.335258 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9820  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:59.162957 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9821  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:59.166446 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9822  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:59.167199 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9823  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:14:59.993680 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9824  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:14:59.996955 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9825  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:14:59.997712 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9826  job 892406418122309633: updated with schema change for table 33, mutation 0
I230819 02:15:00.821601 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9827  job 892406418867224577: updated with schema change for table 51, mutation 0
I230819 02:15:00.824955 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9828  job 892406418116804609: updated with schema change for table 4, mutation 0
I230819 02:15:00.825710 44197 sql/table.go:227  [T1,Vsystem,n1,client=127.0.0.1:59756,hostssl,user=root] 9829  job 892406418122309633: updated with schema change for table 33, mutation 0
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/release-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSystemPrivilegesUserIDMigration1500Users.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30761

--- a/pkg/upgrade/upgrades/system_privileges_user_id_migration_test.go
+++ b/pkg/upgrade/upgrades/system_privileges_user_id_migration_test.go
@@ -21,6 +21,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
@@ -50,6 +51,9 @@ func runTestSystemPrivilegesUserIDMigration(t *testing.T, numUsers int) {
 	tc := testcluster.StartTestCluster(t, 1 /* nodes */, base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
 			Knobs: base.TestingKnobs{
+				SQLEvalContext: &eval.TestingKnobs{
+					ForceProductionValues: true,
+				},
 				Server: &server.TestingKnobs{
 					DisableAutomaticVersionUpgrade: make(chan struct{}),
 					BootstrapVersionKeyOverride:    clusterversion.V22_2,

